builder/googlecompute/driver_mock.go

```diff
@@ -117,17 +119,23 @@ func (d *DriverMock) CreateImage(name, description, family, zone, disk string, i
 	if d.CreateImageResultSizeGb == 0 {
 		d.CreateImageResultSizeGb = 10
 	}
-
+	imageFeatures := make([]*compute.GuestOsFeature, len(image_features))
+	for _, v := range image_features {
+		imageFeatures = append(imageFeatures, &compute.GuestOsFeature{
+			Type: v,
+		})
+	}
 	resultCh := d.CreateImageResultCh
 	if resultCh == nil {
 		ch := make(chan *Image, 1)
 		ch <- &Image{
-			Labels:    d.CreateImageLabels,
-			Licenses:  d.CreateImageLicenses,
-			Name:      name,
-			ProjectId: d.CreateImageResultProjectId,
-			SelfLink:  d.CreateImageResultSelfLink,
-			SizeGb:    d.CreateImageResultSizeGb,
+			Labels:          d.CreateImageLabels,
+			Licenses:        d.CreateImageLicenses,
+			GuestOsFeatures: imageFeatures,
```

Comment:
The other elements here look to be sorted alphanumeric:

```suggestion
			GuestOsFeatures: imageFeatures,
			Labels:          d.CreateImageLabels,
			Licenses:        d.CreateImageLicenses,
		
```

Reply:
Hello,

Thank you for the review! Please try me fix it!

Reply:
Hello,

Thank you for the review!

Please take a look 3fd0bb0:

    * Fixed GuestOsFeatures list
    * Added test to verify that the slice is correctly created.
    * Fixed order to be sorted alphanumeric

Thank you!